From bd7acf53a3fe749dd4ffd6fee7c3d1b6e7b24631 Mon Sep 17 00:00:00 2001
From: Vinod Kumar <vinod.kumarx.vinod.kumar@intel.com>
Date: Wed, 20 Sep 2017 12:40:23 +0800
Subject: [PATCH 018/102] ASoC: Intel: Skylake: Removed the I2S 5 BCLK multiple
 configuration

I2S5 BCLK Pin was configuring during the enablment of I2S 6 and I2S 5.
This patch contains the following changes:
- Removed the I2S 5 BLCK entry for I2S 6.
- Removed the I2S 1 MCLK Pin configuration as not required.
- Optimized the code by reducing allocated memory region.

Signed-off-by: Vinod Kumar <vinod.kumarx.vinod.kumar@intel.com>
---
 sound/soc/intel/boards/apli_lhcrb_generic.c | 14 ++++++--------
 1 file changed, 6 insertions(+), 8 deletions(-)

diff --git a/sound/soc/intel/boards/apli_lhcrb_generic.c b/sound/soc/intel/boards/apli_lhcrb_generic.c
index fe89160..864b74c 100644
--- a/sound/soc/intel/boards/apli_lhcrb_generic.c
+++ b/sound/soc/intel/boards/apli_lhcrb_generic.c
@@ -32,8 +32,8 @@
 #define WEST_BASE 0xd0c70000
 #define WEST_GPIO_VALUE 0x44000800
 
-/* NORTHWEST - GPIO 74 */
-#define I2S_1 0x610
+/* NORTHWEST - GPIO 75 */
+#define I2S_1 0x618
 /* NORTHWEST - GPIO 89 */
 #define I2S_3 0x688
 /* WEST - GPIO 150*/
@@ -122,11 +122,11 @@ static int apli_ssp_northwest_gpio_init(struct snd_soc_pcm_runtime *rtd)
 	switch (rtd->dai_link->be_id) {
 	case 0:
 		gpio_addr = (void *)ioremap_nocache(NORTHWEST_BASE
-				+ I2S_1, 0x30);
+				+ I2S_1, 0x20);
 		break;
 	case 2:
 		gpio_addr = (void *)ioremap_nocache(NORTHWEST_BASE
-				+ I2S_3, 0x30);
+				+ I2S_3, 0x20);
 		break;
 	default:
 		return -EINVAL;
@@ -141,7 +141,6 @@ static int apli_ssp_northwest_gpio_init(struct snd_soc_pcm_runtime *rtd)
 	memcpy_toio(gpio_addr + 0x8, &gpio_value, sizeof(gpio_value));
 	memcpy_toio(gpio_addr + 0x10, &gpio_value, sizeof(gpio_value));
 	memcpy_toio(gpio_addr + 0x18, &gpio_value, sizeof(gpio_value));
-	memcpy_toio(gpio_addr + 0x20, &gpio_value, sizeof(gpio_value));
 
 	iounmap(gpio_addr);
 	return 0;
@@ -161,10 +160,10 @@ static int apli_ssp_west_gpio_init(struct snd_soc_pcm_runtime *rtd)
 
 	switch (rtd->dai_link->be_id) {
 	case 4:
-		gpio_addr = (void *)ioremap_nocache(WEST_BASE + I2S_5, 0x30);
+		gpio_addr = (void *)ioremap_nocache(WEST_BASE + I2S_5, 0x20);
 		break;
 	case 5:
-		gpio_addr = (void *)ioremap_nocache(WEST_BASE + I2S_6, 0x30);
+		gpio_addr = (void *)ioremap_nocache(WEST_BASE + I2S_6, 0x20);
 		break;
 	default:
 		return -EINVAL;
@@ -179,7 +178,6 @@ static int apli_ssp_west_gpio_init(struct snd_soc_pcm_runtime *rtd)
 	memcpy_toio(gpio_addr + 0x8, &gpio_value, sizeof(gpio_value));
 	memcpy_toio(gpio_addr + 0x10, &gpio_value, sizeof(gpio_value));
 	memcpy_toio(gpio_addr + 0x18, &gpio_value, sizeof(gpio_value));
-	memcpy_toio(gpio_addr + 0x20, &gpio_value, sizeof(gpio_value));
 
 	iounmap(gpio_addr);
 	return 0;
-- 
1.9.1

